scripts/imageProcessing/sliceImages.js

```diff
@@ -0,0 +1,104 @@
+const fs = require('fs');
+const { execSync } = require('child_process');
+
+const universalFrameSize = 64;
+const universalSheetWidth = 832;
+
+function cropToSubSheet(sheet, width, height, offsetX, offsetY, targetDir) {
+    const index = sheet.lastIndexOf("\/");
+    var directory = "";
+    if (index > -1) {
+        directory = sheet.substring(0, index);
+    }
+    if (directory == "") {
+        throw Error("wrong directory")
+    } 
+    const newFile = `${directory}/${targetDir}${sheet.substring(index)}`;
+    execSync(`magick ${sheet} -crop ${width}x${height}+${offsetX}+${offsetY} ${newFile}`, (err) => {
+        if (err) {
+            throw err;
+        }
+    });
+}
+
+function cropSheet(sheet) {
+    cropToSubSheet(sheet, 7*universalFrameSize, 4*universalFrameSize,0,0,"spellcast");
```

Comment:
Let's refactor this. the only things that change per line are the number of frames and the name of the sheet.

So create a map:

```
const animToFramesMap = {
  "spellcast": 7,
  "thrust: 8,
  ...
}
```

Then:

```
Object.entries(animToFramesMap).forEach(([anim, frames] =>
  cropToSubSheet(sheet, frames*universalFrameSize, 0, 4,*universalFramesSize, anim));
```

Reply:
Actually, you do something like this already in chargen.js

Reply:
> Let's refactor this. the only things that change per line are the number of frames and the name of the sheet.

That is not true here, also the height (amount of animation rows), offSetX and offsetY is different per animation.